### PR TITLE
Added link to vendor.md in go.mod

### PR DESCRIPTION
**Pull request by jqmichael**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
This PR pointed people to the right direction if they want to update vendor dependency. 

Tests run.
```
/hack/update-vendor.sh
```

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
NONE
```

### CODE REVIEW

**Comment by BenTheElder** [comment]
`n/a` in the PR body doesn't work, please follow the PR template with the "release-note" code block type with `NONE`

**Comment by BenTheElder** [comment]
the project has automation that reads these

**Comment by jqmichael** [comment]
> `n/a` in the PR body doesn't work, please follow the PR template with the "release-note" code block type with `NONE`

Thanks for the reminder!

File: hack/update-vendor.sh [hunk 1]
```diff
@@ -309,6 +309,8 @@ fi
 kube::log::status "go.mod: adding generated comments"
 add_generated_comments "
 // This is a generated file. Do not edit directly.
+// Ensure you've carefully read
+// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/vendor.md
```
Comment (cblecker) [inline hack/update-vendor.sh:313]:
```suggestion
// https://git.k8s.io/community/contributors/devel/sig-architecture/vendor.md
```

Comment (jqmichael) [reply hack/update-vendor.sh]: Thanks for pointing out. Didn't know the domain name before. Updated.

**Pull request merged**